### PR TITLE
Implement the Scan and driver.Value SQL interfaces

### PR DESCRIPTION
**Pull request by svanharmelen**
It would be nice if the `Version` type can be used together with the `database/sql` package.

### CODE REVIEW

File: version.go [hunk 1]
```diff
@@ -405,3 +406,25 @@ func (v *Version) UnmarshalText(b []byte) error {
...
+		*v = *src
+		return nil
+	case []byte:
+		return v.UnmarshalText(src)
+	case string:
+		return v.UnmarshalText([]byte(src))
+	case nil:
+		return nil
+	default:
+		return fmt.Errorf("Cannot convert %T to Version", src)
```
Comment (radeksimko) [inline version.go:423]:
```suggestion
		return fmt.Errorf("cannot scan %T as Version", src)
```

File: version.go [hunk 2]
```diff
@@ -405,3 +406,25 @@ func (v *Version) UnmarshalText(b []byte) error {
...
+// Scan implements the sql.Scanner interface.
+func (v *Version) Scan(src interface{}) error {
+	switch src := src.(type) {
+	case *Version:
+		*v = *src
+		return nil
+	case []byte:
+		return v.UnmarshalText(src)
+	case string:
+		return v.UnmarshalText([]byte(src))
```
Comment (radeksimko) [inline version.go:419]: Could you please provide any rationale behind the choice of types to support here? Is this based on how different DB engines and Go drivers encode the data? If so - can you link to any of those?

I'm not particularly opposed to any of them but want to understand the scope and intentions here. My - perhaps wrong - assumption is that as long as the `Value()` method below is used, then we may only expect `string` here and not bytes or native `*Version` type?

File: version.go (hunk 1 above)
Comment (svanharmelen) [reply version.go]: Update the text according to your suggestion 👍🏻

File: version.go (hunk 2 above)
Comment (svanharmelen) [reply version.go]: I think I got a little bit carried away here 😏 Removed all but the `string` and `nil` cases 👍🏻

**Comment by svanharmelen** [comment]
@radeksimko made some changes, ready for review again...

File: version.go [hunk 3]
```diff
@@ -405,3 +406,20 @@ func (v *Version) UnmarshalText(b []byte) error {
...
+
+// Scan implements the sql.Scanner interface.
+func (v *Version) Scan(src interface{}) error {
+	switch src := src.(type) {
+	case string:
+		return v.UnmarshalText([]byte(src))
+	case nil:
+		return nil
+	default:
+		return fmt.Errorf("Cannot scan %T as Version", src)
```
Comment (radeksimko) [inline version.go:418]:
```suggestion
		return fmt.Errorf("cannot scan %T as Version", src)
```
https://google.github.io/styleguide/go/decisions.html#error-strings

**Review by radeksimko — APPROVED** [review]

File: version.go (hunk 3 above)
Comment (svanharmelen) [reply version.go]: Yeah I know, but I was following the style used in the rest of this package 😉

**Pull request merged**